addons/account/models/account_move.py

```diff
@@ -1203,8 +1203,11 @@ def date_key(move):
                     # Either the move was posted before, or the name already matches the date.
                     # We can skip recalculating the name when either
                     # - the move already has a name, or
-                    # - the move has no name, but is in a period with other moves (so name should be `/`)
-                    if move_has_name and move.posted_before or not move_has_name and move._get_last_sequence(lock=False):
+                    # - the move has no name, but is in a period with other moves (so name should be `/`), or
+                    # - the move has (temporarily) no date set
+                    if move_has_name and move.posted_before or\
+                        not move_has_name and move._get_last_sequence(lock=False) or\
+                        not move.date:
```

Comment:
```suggestion
                    if (
                        move_has_name and move.posted_before 
                        or not move_has_name and move._get_last_sequence(lock=False) 
                        or not move.date
                    ):
```

https://peps.python.org/pep-0008/#maximum-line-length
> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation. 

https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator
> For decades the recommended style was to break after binary operators. But this can hurt readability in two ways: the operators tend to get scattered across different columns on the screen, and each operator is moved away from its operand and onto the previous line

Reply:
That looks a lot better indeed 👍